pygmt/clib/session.py

```diff
@@ -1738,6 +1738,123 @@ def read_virtualfile(
         dtype = {"dataset": _GMT_DATASET, "grid": _GMT_GRID}[kind]
         return ctp.cast(pointer, ctp.POINTER(dtype))
 
+    def return_dataset(
+        self,
+        output_type: Literal["pandas", "numpy", "file"],
```

Comment:
Should we set a default output type here? It looks like we're using `pandas` as the default in #3092.

```suggestion
        output_type: Literal["pandas", "numpy", "file"] = "pandas",
```

Reply:
It makes no differences because we always call the function with the `output_type` parameter, e.g.,:
```
        return lib.return_dataset(
            output_type=output_type,
            vfile=vouttbl,
        )
```

Reply:
Yes, it doesn't make any difference in the PyGMT modules, but this is a good central location to document that `output_type="pandas"` is the default output (though in #1318, it seemed like most of us were in favour of `output_type="input"` or auto as the default).

Reply:
`output_type="input"` or `auto` may not make sense for PyGMT, especially in cases like:

1. the input data is a file, then `auto` means outputting to a file by default, then `outfile` is required.
2. the input data is vectors (e.g., `x`/`y`/`z`) and each vector can be a list/ndarray/pd.Series. Then what's the expected format if `auto`/`input` is used?

Reply:
Yeah, not saying that `output_type="auto"` would be easy to implement :slightly_smiling_face: I think the default `output_type="pandas"` is fine for now, and we can discuss more about what the ideal output type would be in #1318 (if there is still any debate that needs to be had).